File: BackEnd-Siukat/models/log_ayah.go
package models

import "time"

type LogAyah struct {
	IDLogAyah        int        `gorm:"primaryKey;column:id_log_ayah" json:"id_log_ayah"`
	NoPeserta        string     `gorm:"column:no_peserta;type:varchar(255)" json:"no_peserta"`
	StatusAyah       string     `gorm:"column:status_ayah;type:enum('hidup','wafat')" json:"status_ayah"`
	NamaAyah         string     `gorm:"column:nama_ayah;type:varchar(255)" json:"nama_ayah"`
	NikAyah          string     `gorm:"column:nik_ayah;type:varchar(255)" json:"nik_ayah"`
	TeleponAyah      string     `gorm:"column:telepon_ayah;type:varchar(255)" json:"telepon_ayah"`
	AlamatAyah       string     `gorm:"column:alamat_ayah;type:text" json:"alamat_ayah"`
	ProvinsiAyah     string     `gorm:"column:provinsi_ayah;type:varchar(255)" json:"provinsi_ayah"`
	KabkotAyah       string     `gorm:"column:kabkot_ayah;type:varchar(255)" json:"kabkot_ayah"`
	KecamatanAyah    string     `gorm:"column:kecamatan_ayah;type:varchar(255)" json:"kecamatan_ayah"`
	PekerjaanAyah    int        `gorm:"column:pekerjaan_ayah" json:"pekerjaan_ayah"`
	PenghasilanAyah  int        `gorm:"column:penghasilan_ayah" json:"penghasilan_ayah"`
	SampinganAyah    int        `gorm:"column:sampingan_ayah" json:"sampingan_ayah"`
	ScanKtpAyah      string     `gorm:"column:scan_ktp_ayah;type:varchar(255)" json:"scan_ktp_ayah"`
	ScanSlipAyah     string     `gorm:"column:scan_slip_ayah;type:varchar(255)" json:"scan_slip_ayah"`
	TempatLahirAyah  string     `gorm:"column:tempat_lahir_ayah;type:varchar(255)" json:"tempat_lahir_ayah"`
	TanggalLahirAyah *time.Time `gorm:"column:tanggal_lahir_ayah;type:date" json:"tanggal_lahir_ayah"`
	Atribut          string     `gorm:"column:atribut;type:enum('original','sanggah')" json:"atribut"`
	Executor         string     `gorm:"column:executor;type:varchar(255)" json:"executor"`
	Timestamp        *time.Time `gorm:"column:timestamp;type:datetime" json:"timestamp"`
}

func (LogAyah) TableName() string {
	return "log_ayah"
}

File: ZhammDataBeLama/siukat-service-dev_snmptn_2022/models/log_ayah.js
var Sequelize = require('sequelize')
var db = require('../config/database')

const LogAyah = db.define('log_ayah', {
  id_log_ayah:{
  	type: Sequelize.INTEGER,
  	primaryKey: true
  },
  no_peserta: {
    type: Sequelize.STRING
  },
  status_ayah: {
    type: Sequelize.ENUM('hidup', 'wafat')
  },
  nama_ayah: {
    type: Sequelize.STRING
  },
  nik_ayah: {
    type: Sequelize.STRING
  },
  telepon_ayah: {
    type: Sequelize.STRING
  },
  alamat_ayah: {
    type: Sequelize.STRING
  },
  provinsi_ayah: {
    type: Sequelize.INTEGER
  },
  kabkot_ayah: {
    type: Sequelize.INTEGER
  },
  kecamatan_ayah: {
    type: Sequelize.INTEGER
  },
  pekerjaan_ayah: {
    type: Sequelize.CHAR
  },
  penghasilan_ayah: {
    type: Sequelize.INTEGER
  },
  sampingan_ayah: {
    type: Sequelize.INTEGER
  },
  scan_ktp_ayah: {
    type: Sequelize.STRING
  },
  scan_slip_ayah: {
    type: Sequelize.STRING
  },
  tempat_lahir_ayah: {
    type: Sequelize.STRING
  },
  tanggal_lahir_ayah: {
    type: Sequelize.DATEONLY
  },
  atribut: {
    type: Sequelize.ENUM('original', 'sanggah')
  },
  executor: {
    type: Sequelize.STRING
  },
  timestamp: {
    type: Sequelize.DATE
  }
},
 {
  // don't add the timestamp attributes (updatedAt, createdAt)
  timestamps: false,

  // don't delete database entries but set the newly added attribute deletedAt
  // to the current date (when deletion was done). paranoid will only work if
  // timestamps are enabled
  paranoid: true,

  // don't use camelcase for automatically added attributes but underscore style
  // so updatedAt will be updated_at
  underscored: true,

  // disable the modification of tablenames; By default, sequelize will automatically
  // transform all passed model names (first parameter of define) into plural.
  // if you don't want that, set the following
  freezeTableName: true,

  // define the table's name
  tableName: 'log_ayah'
}
);
module.exports = LogAyah
